src/display/annotation_editor_layer.js

```diff
@@ -0,0 +1,1118 @@
+  }
+
+  /**
+   * Commit the content we have in this editor.
+   * @returns {undefined}
+   */
+  commit() {
+    this.disableEditMode();
+    this.#contentHTML = this.editorDiv.innerHTML;
+    this.#content = this.#extractText().trimEnd();
+
+    const style = getComputedStyle(this.div);
+    this.width = parseFloat(style.width);
+    this.height = parseFloat(style.height);
+  }
+
+  /** @inheritdoc */
+  shouldGetKeyboardEvents() {
+    return this.isInEditMode();
+  }
+
+  /**
+   * ondblclick callback.
+   * @param {MouseEvent} event
+   */
+  dblclick(event) {
+    this.enableEditMode();
+    this.editorDiv.focus();
+  }
+
+  /** @inheritdoc */
+  render() {
+    if (this.div) {
+      return this.div;
+    }
+
+    super.render();
+    this.editorDiv = document.createElement("div");
+    this.editorDiv.tabIndex = 0;
+    this.editorDiv.className = "internal";
+    this.editorDiv.contentEditable = true;
+
+    const { style } = this.editorDiv;
+    style.fontSize = `calc(${this.#fontSize}px * var(--zoom-factor))`;
+    style.minHeight = `calc(${1.5 * this.#fontSize}px * var(--zoom-factor))`;
+    style.color = this.#color;
+
+    this.div.appendChild(this.editorDiv);
+
+    this.overlayDiv = document.createElement("div");
+    this.overlayDiv.classList.add("overlay", "enabled");
+    this.div.appendChild(this.overlayDiv);
+
+    // TODO: implement paste callback.
+    // The goal to sanitize and have something suitable for this
+    // editor.
+    bindEvents(this, this.div, ["dblclick"]);
+
+    if (this.width) {
+      // This editor has been created in using copy (ctrl+c).
+      this.setAt(this.x + this.width, this.y + this.height);
+      // eslint-disable-next-line no-unsanitized/property
+      this.editorDiv.innerHTML = this.#contentHTML;
+    }
+
+    return this.div;
+  }
+
+  /** @inheritdoc */
+  serialize() {
+    const rect = this.div.getBoundingClientRect();
+    const [x1, y1] = Util.applyTransform(
+      [this.x, this.y + rect.height],
+      this.parent.viewport.inverseTransform
+    );
+
+    const [x2, y2] = Util.applyTransform(
+      [this.x + rect.width, this.y],
+      this.parent.viewport.inverseTransform
+    );
+
+    return {
+      type: "FreeText",
+      color: [0, 0, 0],
+      fontSize: this.#fontSize,
+      value: this.#content,
+      pageIndex: this.parent.pageIndex,
+      rect: [x1, y1, x2, y2],
+    };
+  }
+}
+
+class AnnotationEditorLayer {
+  #boundClick;
+
+  static _activeEditor = null;
+
+  static _allEditors = new Map();
+
+  static _allowClick = true;
+
+  static _clipboardManager = new ClipboardManager();
+
+  static _commandManager = new CommandManager();
+
+  static _currentEditorClass = null;
+
+  static _currentEditorType = null;
+
+  static _idManager = new IdManager();
+
+  static _isAllSelected = false;
+
+  static _keyboardManager = new KeyboardManager([
+    [["ctrl+a", "mac+meta+a"], AnnotationEditorLayer.prototype.selectAll],
+    [
+      ["ctrl+shift+A", "mac+alt+meta+æ"],
```

Comment:
Do these shortcuts actually make sense?
When using Firefox, on Windows, the "ctrl+shift+A" shortcut will open the `about:addons` page.

Reply:
Acrobat has a shortcut to select/deselect all and I thought it was for newly added annotations but I was wrong.
So I'll remove them.